### ISSUE TITLE
Text inserted on the first line when editor is not in focus

### ISSUE THREAD

**Issue by pvahia** [xdan/jodit#610]
**Jodit Version:** 3.4.xxxxx

**Browser:** Chrome
**OS:** Windows
**Is React App:** False

**Step 1)** Type something in https://xdsoft.net/jodit/play.html
**Step 2)** Position the cursor at the end of the line
**Step 3)** Click anywhere outside of the editor
**Step 4)** Insert a symbol or an image 

**Expected behavior:** The symbol/image should be inserted at the cursor position (i.e. at the end of the line).

**Actual behavior:** The symbol/image gets inserted at the beginning of the editor.

**Comment by Sayan751** [comment]
I am also seeing this issue when inserting text or image using the API; such as `editor.s.insertHTML()`, `editor.s.insertImage()` etc. It seems that with loss of focus, the selection information is lost.

**Comment by pvahia** [comment]
> **Jodit Version:** 3.4.xxxxx
> 
> **Browser:** Chrome
> **OS:** Windows
> **Is React App:** False
> 
> **Step 1)** Type something in https://xdsoft.net/jodit/play.html
> **Step 2)** Position the cursor at the end of the line
> **Step 3)** Click anywhere outside of the editor
> **Step 4)** Insert a symbol or an image
> 
> **Expected behavior:** The symbol/image should be inserted at the cursor position (i.e. at the end of the line).
> 
> **Actual behavior:** The symbol/image gets inserted at the beginning of the editor.

``

**Comment by pvahia** [comment]
> **Jodit Version:** 3.4.xxxxx
> 
> **Browser:** Chrome
> **OS:** Windows
> **Is React App:** False
> 
> **Step 1)** Type something in https://xdsoft.net/jodit/play.html
> **Step 2)** Position the cursor at the end of the line
> **Step 3)** Click anywhere outside of the editor
> **Step 4)** Insert a symbol or an image
> 
> **Expected behavior:** The symbol/image should be inserted at the cursor position (i.e. at the end of the line).
> 
> **Actual behavior:** The symbol/image gets inserted at the beginning of the editor.

Still able to reproduce, any update here?

**Comment by Sayan751** [comment]
@xdan I tried to utilize the `blur` event to save a selection marker, and later before inserting the html or image I tried to restore the marker. But no success there as well. Can you suggest any workaround on that line?

**Comment by Sayan751** [comment]
Figured it out! My approach of saving the marker and later restoring those worked.

It goes something like this.

```typescript
private saveSelectionMarker(event: Event) {
  const editor = this.editor;
  if (!editor.editor.contains(event.target as Node)) { return; }
  const selection = editor.selection;
  const _markerInfo = this.markerInfo;
  this.markerInfo = selection.save();

  if (_markerInfo === null) { return; }

  for (const info of _markerInfo) {
    selection.removeNode(document.querySelector(`#${info.startId}`));
    if (info.endId) {
      selection.removeNode(document.querySelector(`#${info.endId}`));
    }
  }
}	

private restoreSelection() {
  const info = this.markerInfo;
  if (info !== null) {
    this.editor.selection.restore(info);
  }
  this.markerInfo = null;
}

// add event handler
this.editor.editor.addEventListener('click', this.saveSelectionMarker.bind(this), { capture: true });

// inserting HTML
this.restoreSelection();
this.editor.s.insertHTML('<span>foo</span>');
```

**Comment by sid-ihycq** [comment]
In my case, the editor always scroll to the top, even through the forcus is at lastest place.  it seems that selection.save/restore can only hold focus at the right place. Is there any way to keep the scroll position? Tks! @xdan

**Comment by xdan** [comment]
Fixed in 3.6.13
